### PR TITLE
Testing clear-all , editing and check-status functions

### PR DESCRIPTION
**Pull request by Fikerte-T**
In this branch:

Create tests for `editTask()` function.
Create tests for `removeTasks()` function.
Create tests for `taskCompleted()` function.

### CODE REVIEW

File: src/__mocks__/mockItem.js [hunk 1]
```diff
@@ -45,5 +45,100 @@ const addTask = () => {
...
+    { id: 2, description: 'item 2', complete: true },
+    { id: 3, description: 'item 3', complete: false },
+  ];
+  // const checked = document.querySelectorAll('input[type="checkbox"]');
+  // checked.forEach((checkbox) => {
+  //   if (checkbox.checked) {
+  //     checkbox.parentElement.remove();
+  //   }
+  // });
+  // eslint-disable-next-line no-plusplus
```
Comment (mikenath223) [inline src/__mocks__/mockItem.js:94]: - [ ] Please remove this commented code.

File: src/__mocks__/mockItem.js [hunk 2]
```diff
@@ -45,5 +45,100 @@ const addTask = () => {
...
+        let i = 0;
+        while (i < items.length) {
+          if (items[i].id > item.id) {
+            // eslint-disable-next-line no-plusplus
+            items[i].id--;
+          }
+          // eslint-disable-next-line no-plusplus
+          i++;
+        }
+        // localStorage.setItem('items', JSON.stringify(items));
```
Comment (mikenath223) [inline src/__mocks__/mockItem.js:109]: - [ ] Remove the commented code.

File: src/__mocks__/mockItem.js [hunk 3]
```diff
@@ -45,5 +45,100 @@ const addTask = () => {
...
+  return items;
+};
+
+const editContent = (string) => {
+  const items = [
+    { id: 1, description: 'item 1', complete: true },
+    { id: 2, description: 'item 2', complete: true },
+    { id: 3, description: 'item 3', complete: false },
+  ];
+  // const edit = e.target;
```
Comment (mikenath223) [inline src/__mocks__/mockItem.js:124]: - [ ] Remove the commented code.

File: src/__mocks__/mockItem.js [hunk 4]
```diff
@@ -45,5 +45,100 @@ const addTask = () => {
...
+  inputEvent.removeAttribute('readonly');
+  inputEvent.value = string;
+
+  // edit.setAttribute('readonly', 'readonly');
+  // inputEvent.value = p.value;
```
Comment (mikenath223) [inline src/__mocks__/mockItem.js:129]: - [ ] Remove the commented code.

File: src/app.test.js [hunk 5]
```diff
@@ -30,3 +34,38 @@ describe('should add an item', () => {
...
 });
+
+describe('should check status', () => {
+  test('status should be true', () => {
+    expect(taskCompleted.taskCompleted()).toEqual([
+      { id: 1, description: 'item 1', complete: true },
+      { id: 2, description: 'item 2', complete: false },
+      { id: 3, description: 'item 3', complete: false },
+    ]);
+  });
```
Comment (mikenath223) [inline src/app.test.js:45]: - [ ] Please add test for clear all completed function.

File: src/mockItem.js [hunk 6]
```diff
@@ -62,3 +62,29 @@ const addTask = (e, items, input, itemsContainer, Item) => {
...
+        items[i].completed = true;
+        localStorage.setItem('items', JSON.stringify(items));
+      }
+    }
+  } else {
+    complete.parentElement.classList.remove('completed');
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < items.length; i++) {
+      // eslint-disable-next-line eqeqeq
+      if (complete.parentNode.parentNode.id == items[i].id) {
```
Comment (mikenath223) [inline src/mockItem.js:84]: - [ ] Kindly remove all the eslint disable rules its best to fix the errors.

**Comment by mikenath223** [comment]
- [ ] Also include in the readme instructions to run tests.

**Review by wandji20 — APPROVED** [review]
## Hi @Lordkaito  :raised_hands:

##### You have done exceptionally well, and have met all requirements for this project :+1:

# Status: Approved 👍 :partying_face: 
Your project has been approved, however, I have left some comments to help improve your 
## Congratulations :+1:  :fireworks: 
## Optional
- [ optional ] It will be nice to ignore all files that are not relevant to run this project(eg .lintrc, .stylelintrc.json, .eslintrc.json etc).

_Go ahead and merge this branch. Also, don't forget to submit the project completion form.
Good luck with your next project :mechanical_arm:  :slightly_smiling_face:_ 

 If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**